### PR TITLE
Add asp core support for signalr

### PR DESCRIPTION
**Pull request by oydevaas**

### CODE REVIEW

File: TypeGap/TypeFluent.cs [hunk 1]
```diff
@@ -30,6 +30,7 @@ public class TypeFluent
         private GapEnumGenerator _enumGenerator = new RegularEnumGenerator(false, EnumValueMode.Number);
         private string _indent = "    ";
         private ITsModelVisitor _modelVisitor;
+        private Dictionary<Type, string> _typeConversions;
```
Comment (rstarkov) [inline TypeGap/TypeFluent.cs]: These changes shouldn't be in the PR, can you pull all commits from master and rebase your PR please?

File: TypeGap/SignalRGenerator.cs [hunk 2]
```diff
@@ -38,11 +39,13 @@ public string HubSignalRName
...
     {
         internal const string HUB_TYPE = "Microsoft.AspNet.SignalR.Hub";
+        internal const string HUB_TYPE_CORE = "Microsoft.AspNetCore.SignalR.Hub";
 
         private string GenerateHubs(Assembly assembly, TypeConverter converter)
         {
             var hubs = assembly.GetTypes()
-                .Where(t => t.GetDnxCompatible().BaseType != null && t.GetDnxCompatible().BaseType.FullName != null && t.GetDnxCompatible().BaseType.FullName.Contains(HUB_TYPE))
+                .Where(t => t.GetDnxCompatible().BaseType != null && t.GetDnxCompatible().BaseType.FullName != null &&
+                            t.GetDnxCompatible().BaseType.FullName.Contains(HUB_TYPE) || t.GetDnxCompatible().BaseType.FullName.Contains(HUB_TYPE_CORE))
```
[the remark this answers is not in the document]
Comment (oydevaas) [reply TypeGap/SignalRGenerator.cs]: true

File: TypeGap/SignalRGenerator.cs [hunk 3]
```diff
@@ -26,7 +26,8 @@ public string HubSignalRName
         {
             get
             {
-                var attribute = HubType.GetDnxCompatible().GetCustomAttributes().Where(ca => ca.GetType().FullName == "Microsoft.AspNet.SignalR.Hubs.HubNameAttribute").FirstOrDefault();
+                var attribute = HubType.GetDnxCompatible().GetCustomAttributes().Where(ca =>
+                    ca.GetType().Name == "HubNameAttribute").FirstOrDefault();
```
[the remark this answers is not in the document]
Comment (oydevaas) [reply TypeGap/SignalRGenerator.cs]: The problem here is that ASP Core don't provide such a attribute name.
This name is custom made in my project. So ASP.NET provide `some_namespace.HubName` but ASP.CORE doesn't.

File: TypeGap/SignalRGenerator.cs [hunk 4]
```diff
@@ -147,10 +151,10 @@ private void GenerateHubInterfaces(SignalRHubDesc hub, ScriptBuilder scriptBuild
...
             // Build the interface containing the SERVER methods
             scriptBuilder.AppendLineIndented($"interface I{hub.HubClassName} {{");
-            using (scriptBuilder.IncreaseIndentation())
-            {
-                GenerateMethods(scriptBuilder, hub, converter, false, (mi, tc) => GenerateMethodDeclaration(mi, tc, hub, false));
-            }
+            //using (scriptBuilder.IncreaseIndentation())
+            //{
+            //    GenerateMethods(scriptBuilder, hub, converter, false, (mi, tc) => GenerateMethodDeclaration(mi, tc, hub, false));
+            //}
```
[the remark this answers is not in the document]
Comment (oydevaas) [reply TypeGap/SignalRGenerator.cs]: good idea

File: TypeGap/TypeFluent.cs (hunk 1 above)
Comment (oydevaas) [reply TypeGap/TypeFluent.cs]: I will revert those

**Pull request merged**